### PR TITLE
UCA: Uncomment UCA variables in defaults, fix consequent breakage

### PR DESCRIPTION
**Pull request by mcv21**
The Ubuntu Cloud Archive-related (UCA) defaults in
roles/ceph-defaults/defaults/main.yml were commented out, which means
if you set `ceph_repository` to "uca", you get undefined variable
errors, e.g.

```
The task includes an option with an undefined variable. The error was: 'ceph_stable_repo_uca' is undefined

The error appears to have been in '/nfs/users/nfs_m/mv3/software/ceph-ansible/roles/ceph-common/tasks/installs/debian_uca_repository.yml': line 6, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

- name: add ubuntu cloud archive repository
  ^ here

```

Unfortunately, uncommenting these results in some other breakage,
because further roles were written that use the fact of
`ceph_stable_release_uca` being defined as a proxy for "we're using
UCA", so try and install packages from the bionic-updates/queens
release, for example, which doesn't work. So there are a few `apt` tasks
that need modifying to not use `ceph_stable_release_uca` unless
`ceph_origin` is `repository` and `ceph_repository` is `uca`.

Closes: #3745
Signed-off-by: Matthew Vernon <mv3@sanger.ac.uk>

### CODE REVIEW

**Comment by mcv21** [comment]
The PR pipeline fails complaining I need to run `generate_group_vars_sample.sh`, but doing so makes no difference (per `git status`); I think this is because the group_vars haven't changed - I just uncommented a couple of variables that were already (incorrectly) commented out.

File: roles/ceph-mds/tasks/non_containerized.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   apt:
     name: ceph-mds
     state: "{{ (upgrade_ceph_packages|bool) | ternary('latest','present') }}"
-    default_release: "{{ ceph_stable_release_uca | default('') }}{{ ansible_distribution_release ~ '-backports' if ceph_origin == 'distro' and ceph_use_distro_backports else '' }}"
+    default_release: "{{ ceph_stable_release_uca | default('') if ceph_origin == 'repository' and ceph_repository == 'uca' else ''}}{{ ansible_distribution_release ~ '-backports' if ceph_origin == 'distro' and ceph_use_distro_backports else ''}}"
```
Comment (guits) [inline roles/ceph-mds/tasks/non_containerized.yml]:
```suggestion
    default_release: "{{ ceph_stable_release_uca | default('') if ceph_origin == 'repository' and ceph_repository == 'uca' else '' }}{{ ansible_distribution_release ~ '-backports' if ceph_origin == 'distro' and ceph_use_distro_backports else '' }}"
```

File: roles/ceph-common/tasks/installs/install_debian_packages.yml [hunk 2]
```diff
@@ -4,6 +4,6 @@
     name: "{{ debian_ceph_pkgs | unique }}"
     update_cache: no
     state: "{{ (upgrade_ceph_packages|bool) | ternary('latest','present') }}"
-    default_release: "{{ ceph_stable_release_uca | default('') }}{{ ansible_distribution_release ~ '-backports' if ceph_origin == 'distro' and ceph_use_distro_backports else '' }}"
+    default_release: "{{ ceph_stable_release_uca | default('') if ceph_origin == 'repository' and ceph_repository == 'uca' else ''}}{{ ansible_distribution_release ~ '-backports' if ceph_origin == 'distro' and ceph_use_distro_backports else ''}}"
```
Comment (guits) [inline roles/ceph-common/tasks/installs/install_debian_packages.yml]:
```suggestion
    default_release: "{{ ceph_stable_release_uca | default('') if ceph_origin == 'repository' and ceph_repository == 'uca' else '' }}{{ ansible_distribution_release ~ '-backports' if ceph_origin == 'distro' and ceph_use_distro_backports else '' }}"
```

File: roles/ceph-defaults/defaults/main.yml [hunk 3]
```diff
@@ -202,9 +202,9 @@ ceph_rhcs_cdn_debian_repo_version: "/3-release/" # for GA, later for updates use
 # usually has newer Ceph releases than the normal distro repository.
 #
 #
-#ceph_stable_repo_uca: "http://ubuntu-cloud.archive.canonical.com/ubuntu"
-#ceph_stable_openstack_release_uca: queens
-#ceph_stable_release_uca: "{{ansible_lsb.codename}}-updates/{{ceph_stable_openstack_release_uca}}"
+ceph_stable_repo_uca: "http://ubuntu-cloud.archive.canonical.com/ubuntu"
+ceph_stable_openstack_release_uca: queens
+ceph_stable_release_uca: "{{ansible_lsb.codename}}-updates/{{ceph_stable_openstack_release_uca}}"
```
Comment (guits) [inline roles/ceph-defaults/defaults/main.yml]:
```suggestion
ceph_stable_release_uca: "{{ ansible_lsb.codename }}-updates/{{ ceph_stable_openstack_release_uca }}"
```

**Comment by guits** [comment]
@mcv21 not sure what's your issue with the generate script, I've just tried with your branch and seems to be working

**Comment by guits** [comment]
@mcv21 not sure we really wanna see :

`Can this be backported to (at least) stable-3.2, please?`

in the commit message/ git log history.

If you think this patch should be backported, just ask us in the pull request, we will apply the label so mergify will create the backport in the corresponding stable branch once your patch is merged in master.

Thanks!

File: roles/ceph-mds/tasks/non_containerized.yml (hunk 1 above)
Comment (mcv21) [reply roles/ceph-mds/tasks/non_containerized.yml]: Updated with this change.

File: roles/ceph-common/tasks/installs/install_debian_packages.yml (hunk 2 above)
Comment (mcv21) [reply roles/ceph-common/tasks/installs/install_debian_packages.yml]: Updated with this change.

**Comment by mcv21** [comment]
@guits I've been away, hence the slightly slow reply. Apologies!

I've updated the PR with your suggestions above (and the generate script now does what I expect, since the defaults file has changes other than just uncommenting).

Noting here I'd like the backport to stable-3.2, having taking it out of the commit message (and PR text).

File: roles/ceph-defaults/defaults/main.yml (hunk 3 above)
Comment (mcv21) [reply roles/ceph-defaults/defaults/main.yml]: Updated with this change.

**Comment by mcv21** [comment]
Hi @guits ,
Do you need anything more from me to get this PR merged?
Thanks :)

**Comment by guits** [comment]
@mcv21 i'm waiting for the CI to be passing

**Pull request merged**